### ISSUE TITLE
Wrong numbers using Datadog.Trace 0.8.0-beta

### ISSUE THREAD

**Issue by vadimi** [DataDog/dd-trace-dotnet#293]
**Describe the bug**
Ever since we switched to `Datadog.Trace` version `0.8.0-beta` we're getting wrong tracing results in dotnet core apps in docker containers. The numbers are much higher than they actually are. I was able to debug it a little bit and found that `TraceContext` was changed to use `_utcStart.AddTicks(_stopwatch.ElapsedTicks)` instead of `_start.Add(_sw.Elapsed)` in version `0.7.1-beta`.

I wrote a simple app to reproduce the issue. This is what it does:

```csharp
static async Task Main(string[] args)
{
    var utcStart = DateTimeOffset.UtcNow;
    var stopwatch = Stopwatch.StartNew();
    await Task.Delay(100);
    var diff = utcStart.AddTicks(stopwatch.ElapsedTicks) - utcStart;
    Console.WriteLine("{0}", stopwatch.ElapsedTicks);
    Console.WriteLine("diff using ticks: {0}ms", diff.TotalMilliseconds);
    Console.WriteLine("diff using offset: {0}ms", (DateTimeOffset.UtcNow - utcStart).TotalMilliseconds);
}
```

Here are results of running this app on **Linux**:

```
dotnet run
101802724
diff using ticks: 10179.6821ms
diff using offset: 114.2126ms
```

**Windows**

```
dotnet run
293109
diff using ticks: 29.309ms
diff using offset: 110.9422ms
```

windows results are also kind of weird.

**Runtime environment (please complete the following information):**
- Instrumentation mode: manual
- Tracer version: [e.g. 0.8.0]
- OS: Linux, docker, kubernetes
- CLR: .NET Core 2.1

**Comment by vadimi** [comment]
Based on https://docs.microsoft.com/en-us/windows/desktop/sysinfo/acquiring-high-resolution-time-stamps#resolution-precision-accuracy-and-stability calculating duration like that works fine in Windows and Linux:

```cs
var start = Stopwatch.GetTimestamp();
await Task.Delay(100);
var end = Stopwatch.GetTimestamp() - start;
double elapsedMs = end * (1000.0 / Stopwatch.Frequency);
```

**Comment by lucaspimentel** [comment]
Thanks for reporting this. We'll look into it right away.

**Comment by lucaspimentel** [comment]
Interestingly, in [`DateTimeOffset.AddTicks()`](https://docs.microsoft.com/en-us/dotnet/api/system.datetimeoffset.addticks) a tick seems to always be 100 nanoseconds, but the length of ticks in [`Stopwatch.ElapsedTicks`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch.elapsedticks) can vary by platform (see also [`Stopwatch.Frequency`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch.frequency)).

**Comment by lucaspimentel** [comment]
@vadimi: The issue should now be fixed in [0.8.2](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v0.8.2-beta), so I'm closing this issue. Thanks again for reporting it.
